index.js

```diff
@@ -19,11 +19,21 @@ var nodeStatusCodes = require('node-status-codes');
 var isPlainObj = require('is-plain-obj');
 var parseJson = require('parse-json');
 
+function backoff(iter) {
+	var noise = Math.random() * 1000;
+	return (1 << iter) * 1000 + noise;
+}
+
+function isNetworkError(err) {
+	return err.code === 'EAI_AGAIN' || err.code === 'ETIMEDOUT' || err.code === 'ESOCKETTIMEDOUT';
```

Comment:
On second though, we should retry on any error in `error` handler of http.request. There also `ECONNRESET` and `ECONNREFUSED` that will occur, when remote server is restarting.